python/ray/train/trainer.py

```diff
@@ -186,12 +226,22 @@ def _fetch_next_result(self) -> Optional[List[Dict]]:
             first_result = results[0]
             result_type = first_result.type
             if result_type is TrainingResultType.REPORT:
-                result_data = [r.data for r in results]
+                if _use_storage_context():
+                    # TODO(justinvyu): This tuple is what TrainingResult should be.
+                    result_data = [
+                        (r.data, None if rank > 0 else self._checkpoint_to_report)
+                        for rank, r in enumerate(results)
+                    ]
+                else:
+                    result_data = [r.data for r in results]
                 return result_data
             elif result_type is TrainingResultType.CHECKPOINT:
-                self._checkpoint_manager._process_checkpoints(
-                    results, decode_checkpoint_fn=self._backend._decode_data
-                )
+                if _use_storage_context():
+                    self._process_checkpoint_results(results)
```

Comment:
Do we still need to involve the checkpoint manager to delete old checkpoints, etc?

Reply:
The Train checkpoint manager is already disabled on purpose and basically doesn't do anything. Tune's checkpoint manager is the one who actually keeps track of the heap of top K checkpoints and does deletion. I think we should just keep it this way for now, and just fully avoid the train checkpoint manager.

Here's where it will happen:

https://github.com/ray-project/ray/pull/37888/files/ee4ccbd8be5457291a2032826e8a30db4df1f72f#diff-918fa19407dbc59ec621b7fbc9e35cf991659dc3b6657b5f4c46534d8dfb73a3R54-R66